webapp/platform/types/src/message_attachments.ts

```diff
@@ -1,30 +1,141 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import type {PostAction} from './integration_actions';
+import {isPostAction, type PostAction} from './integration_actions';
+import {isArrayOf} from './utilities';
 
 export type MessageAttachment = {
-    id: number;
-    fallback: string;
-    color: string;
-    pretext: string;
-    author_name: string;
-    author_link: string;
-    author_icon: string;
-    title: string;
-    title_link: string;
-    text: string;
-    fields: MessageAttachmentField[];
-    image_url: string;
-    thumb_url: string;
-    footer: string;
-    footer_icon: string;
-    timestamp: number | string;
+    fallback?: string;
+    color?: string;
+    pretext?: string;
+    author_name?: string;
+    author_link?: string;
+    author_icon?: string;
+    title?: string;
+    title_link?: string;
+    text?: string;
+    fields?: MessageAttachmentField[];
+    image_url?: string;
+    thumb_url?: string;
+    footer?: string;
+    footer_icon?: string;
     actions?: PostAction[];
 };
 
+export function isMessageAttachmentArray(v: unknown): v is MessageAttachment[] {
+    return isArrayOf(v, isMessageAttachment);
+}
+
+function isMessageAttachment(v: unknown): v is MessageAttachment {
+    if (typeof v !== 'object' || !v) {
+        return false;
+    }
+
+    if ('fallback' in v && typeof v.fallback !== 'string') {
+        return false;
+    }
+
+    // We may consider adding more validation to what color may be
+    if ('color' in v && typeof v.color !== 'string') {
+        return false;
+    }
+
+    if ('pretext' in v && typeof v.pretext !== 'string') {
+        return false;
+    }
+
+    if ('author_name' in v && typeof v.author_name !== 'string') {
+        return false;
+    }
+
+    // Where it is used, we are calling isUrlSafe. We could consider calling it here
```

Comment:
That being said... should I remove the comments, or better leave them there for future reference?

Reply:
I've been trying to remove that sort of comment from my code since it's kind of a TODO, but I'm 0/5 on if you need to go do that now

Reply:
I'm going to leave these comments in place, since I feel they add insightful documentation to the code for the future. Not sure if will be ever be act on, but still valuable information.